nodestream/databases/neo4j/query_executor.py

```diff
@@ -81,7 +82,16 @@ async def execute(self, query: Query, log_result: bool = False):
                 "uri": self.driver._pool.address.host,
             },
         )
-        await self.driver.verify_connectivity()
+
+        try:
+            await self.driver.verify_connectivity()
+        except ServiceUnavailable:
+            self.logger.exception(
+                "Neo4j Session timed out while waiting for other steps to resolve. Trying one more time.",
```

Comment:
Sorry, final thought on this :). Since there is a variety of issues that could cause this, we should probably have the message be less assertive on the timeout:
```
Neo4j Driver failed to connect for an unspecified reason. Maybe due to waiting for event loop to pass back control. Trying one more time.
```